### PR TITLE
:sparkles:  Rebase Kubernetes 1.31

### PR DESCRIPTION
**Pull request by mjudeikis**
## Summary

In general, this was very smooth sailing. So smooth Im worried I missed something bit :)

Full diff can be found here, where I tool baseline 1.31 and this PR shows diff we currently carry as of 1.31 rebase:
https://github.com/kcp-dev/kubernetes/pull/147/commits 

In general, most fixes were machinery-related:
1. Queue type change in upstream
2. Deprecation of stopCh (our change)
3. Some control plane changes we landed post 1.30 cut
4. Service account changes (getter of service account keys and our changes mostly)

In general a lot of sed/replace type of work. 

Biggest issues still is code-generation, where we are on our own :) 

## Related issue(s)

Fixes #

## Release Notes

```release-note
Rebase 1.31
```

### CODE REVIEW

**Review by embik — CHANGES_REQUESTED** [review]
Couple of nits, but overall looking really solid 👍🏻

File: pkg/cache/server/server.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ func (s *Server) PrepareRun(ctx context.Context) (preparedServer, error) {
 	logger := klog.FromContext(ctx).WithValues("component", "cache-server")
 	if err := s.apiextensions.GenericAPIServer.AddPostStartHook("bootstrap-cache-server", func(hookContext genericapiserver.PostStartHookContext) error {
 		logger := logger.WithValues("postStartHook", "bootstrap-cache-server")
-		if err := bootstrap.Bootstrap(klog.NewContext(goContext(hookContext), logger), s.ApiExtensionsClusterClient); err != nil {
+		if err := bootstrap.Bootstrap(klog.NewContext(ctx, logger), s.ApiExtensionsClusterClient); err != nil {
```
[the remark this answers is not in the document]
Comment (mjudeikis) [reply pkg/cache/server/server.go]: mistake

**Review by embik — APPROVED** [review]
/lgtm

I'd just be very happy about a PR description documenting what had to be done for the 1.31 upgrade :)

**Comment by mjudeikis** [comment]
> I'd just be very happy about a PR description documenting what had to be done for the 1.31 upgrade :)

Added but needed to think what to add as it was so smooth :D

**Review by embik — APPROVED** [review]
/approve

**Pull request merged**